golem-component-service-base/src/service/component.rs

```diff
@@ -152,6 +233,19 @@ pub trait ComponentService<Namespace> {
         component_name: &ComponentName,
         component_type: ComponentType,
         data: Vec<u8>,
+        files: Option<InitialComponentFilesArchiveAndPermissions>,
+        namespace: &Namespace,
+    ) -> Result<Component<Namespace>, ComponentError>;
+
+    // Files must have been uploaded to the blob store before calling this method
+    async fn create_internal(
+        &self,
+        component_id: &ComponentId,
+        component_name: &ComponentName,
+        component_type: ComponentType,
+        data: Vec<u8>,
+        // None signals that files should be reused from the previous version
+        files: Vec<InitialComponentFile>,
```

Comment:
Probably for consistency with update_internal, use `Option<Vec>`

Reply:
This comment was a broken leftover, removed. There is no semantic difference between Some(vec![]) and None here, so I think it makes sense to not have more possible values than needed here.

Reply:
Okay 👍